wordpress-mobile/WordPress-Android
# Prologue: Background update

This PR updates the Prologue screen background to match the design spec (internal ref: pbArwn-193-p2). 

Here's how it looks like:

<img width="300" src="https://user-images.githubusercontent.com/830056/95368716-df60eb00-08ac-11eb-9b35-ae02f4f1d936.gif">

Note: this PR doesn't make the status bar transparent as defined by the design spec. This work will be done separately later on.

The goal here was to make the background look like a single element that gives us an effect of continuity when swiping through the Prologue pages, rather than five separate backgrounds for each page. To achieve that, I first added a parallax effect to the `ViewPager` to make the foreground elements move a bit faster than the background when swiping pages. As of right now, this is the configuration I went with:

- Background: Normal speed
- Title: 1.25x normal speed
- Content: 1.5x normal speed

As for the background itself, I decided to have a single circle image and then size, tint and position multiple copies of it individually for each page. I also experimented with having a single wide image with all the background circles pre-positioned in it and then make it scroll along with the pages, and while that approach was easier to implement, it didn't look good on different screen sizes, as the circles were positioned independently of the available screen space.

To position the circles in each page, I decided to use a `ConstraintLayout` and have two guidelines for each circle to indicate where they would be positioned, both vertically and horizontally. A [guideline](https://developer.android.com/reference/androidx/constraintlayout/widget/Guideline) needs to be bound to one of edges of the layout: vertical guidelines are bound to either the left or the right side, while horizontal guidelines are bound to either the top or the bottom. We then define a fixed distance in `dp` to indicate how far from that edge the guideline is going to be positioned. Finally, we specify how we want a certain circle drawable to be positioned in relation to that guideline. 

The example below demonstrates a horizontal guideline (in blue) bound to the top of the layout (as indicated by its arrow icon pointing up) and distant from it by `60dp`. The bottom of the top-left circle is positioned using this guideline.

<img width="300" src="https://user-images.githubusercontent.com/830056/95368795-facbf600-08ac-11eb-8fb9-63cadbea6487.png">

Since we're using fixed distance values, the circles will always be positioned at the exact same place, independently of the screen size:

|Small|Medium|Large|Wide|Tablet|
|-----|-----|-----|-----|-----|
|<img width="100" src="https://user-images.githubusercontent.com/830056/95371502-aa569780-08b0-11eb-9d5b-e7133a270c69.png">|<img width="100" src="https://user-images.githubusercontent.com/830056/95371501-a9be0100-08b0-11eb-9a8c-85325f772f85.png">|<img width="100" src="https://user-images.githubusercontent.com/830056/95371498-a9256a80-08b0-11eb-9dbf-85d4c0628acb.png">|<img width="200" src="https://user-images.githubusercontent.com/830056/95371485-a591e380-08b0-11eb-8938-bc05fc83b3bb.png">|<img width="200" src="https://user-images.githubusercontent.com/830056/95371505-aaef2e00-08b0-11eb-856e-0a00d9924759.png">|

Note: please, disregard the page content as it has not been optimized for different screen sizes yet.

This is a bit different from what we have on the design spec, where the circles seem to be more visible when there's more screen space available. One way of addressing this would be to make the guidelines use percentages of the width or the height of a layout instead of a fixed distance value. I did experiment with this a bit but the results seemed to be more unpredictable, but I'm totally open to revisit this approach again if necessary.

## To test

1. Open the app.
1. Log out if necessary. 
1. On the Prologue screen, swipe between the pages and confirm that the background looks and behaves as expected.

Make sure to test on devices with different screen sizes/orientations and Android versions.

---

PR submission checklist:
